### PR TITLE
ScanParam `match` binary version

### PR DESCRIPTION
**Pull request by sazzad16**

### CODE REVIEW

File: src/main/java/redis/clients/util/JedisClusterHashTagUtil.java [hunk 1]
```diff
@@ -14,6 +14,10 @@ public static String getHashTag(String key) {
     return extractHashTag(key, true);
   }
 
+  public static boolean isClusterCompliantMatchPattern(byte[] matchPattern) {
```
Comment (HeartSaVioR) [inline src/main/java/redis/clients/util/JedisClusterHashTagUtil.java]: Looks like this is unrelated to the actual changeset. Minimizing changeset helps us avoiding merge conflict.

**Comment by sazzad16** [comment]
Merged into master, 2.10.
